### PR TITLE
GraphQl-45: Product textarea field format

### PR DESCRIPTION
**Pull request by VitaliyBoyko**
### Description
PROTOTYPE OF FIELD FORMAT STRATEGY

### Fixed Issues (if relevant)
1. magento/graphql-ce#45: Enable GraphQL text fields to contain WYSIWYG/Pagebuilder content (as complex object)

### CODE REVIEW

File: app/code/Magento/CatalogGraphQl/Model/Resolver/Product/ProductTextareaAttribute/FormatFactory.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    /**
+     * @param string $formatIdentifier
+     * @param array $data
+     * @return FormatInterface
+     */
+    public function create(string $formatIdentifier, $data = []) : FormatInterface
+    {
+        $formatClassName = 'Magento\CatalogGraphQl\Model\Resolver\Product\ProductTextareaAttribute\\' . ucfirst($formatIdentifier);
```
Comment (VitaliyBoyko) [inline app/code/Magento/CatalogGraphQl/Model/Resolver/Product/ProductTextareaAttribute/FormatFactory.php]: The temporary solution just to show idea.
I'm about to move class name mapping into di.xml .

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/graphql-ce?pullRequest=186) <br/>All committers have signed the CLA.

File: app/code/Magento/CatalogGraphQl/etc/graphql/di.xml [hunk 2]
```diff
@@ -63,6 +63,13 @@
             <argument name="collectionProcessor" xsi:type="object">Magento\Catalog\Model\Api\SearchCriteria\ProductCollectionProcessor</argument>
         </arguments>
     </type>
+    <type name="Magento\CatalogGraphQl\Model\Resolver\Product\ProductTextAttribute\FormatList">
```
Comment (paliarush) [inline app/code/Magento/CatalogGraphQl/etc/graphql/di.xml]: Looks like unused.

File: app/code/Magento/CatalogGraphQl/Model/Resolver/Product/ProductTextAttribute.php [hunk 3]
```diff
@@ -7,17 +7,17 @@
...
 use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Catalog\Model\Product;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
 use Magento\Catalog\Helper\Output as OutputHelper;
 
 /**
  * Resolve rendered content for attributes where HTML content is allowed
  */
-class ProductHtmlAttribute implements ResolverInterface
+class ProductTextAttribute implements ResolverInterface
```
Comment (paliarush) [inline app/code/Magento/CatalogGraphQl/Model/Resolver/Product/ProductTextAttribute.php]: How about `ProductTextAttribute` => `ProductComplextTextAttribute`?

**Review by paliarush — APPROVED** [review]

File: app/code/Magento/CatalogGraphQl/Model/Resolver/Product/ProductTextAttribute.php (hunk 3 above)
Comment (VitaliyBoyko) [reply app/code/Magento/CatalogGraphQl/Model/Resolver/Product/ProductTextAttribute.php]: Yep, agree.

**Review by paliarush — APPROVED** [review]

File: app/code/Magento/GraphQl/Model/Resolver/ComplexTextValue/HtmlFormat.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * @author Atwix Team
```
Comment (paliarush) [inline app/code/Magento/GraphQl/Model/Resolver/ComplexTextValue/HtmlFormat.php]: Please change the copyright

**Review by paliarush — APPROVED** [review]

**Review by paliarush — APPROVED** [review]

**Review by naydav — APPROVED** [review]

**Review by paliarush — APPROVED** [review]

**Pull request merged**